Source/player.cpp

```diff
@@ -3143,7 +3142,7 @@ StartPlayerKill(int pnum, int earflag)
 			drawhpflag = true;
 
 			if (pcurs >= CURSOR_FIRSTITEM) {
-				DeadItem(player, player.HoldItem, { 0, 0 });
+				DeadItem(player, std::move(player.HoldItem), { 0, 0 });

```

Comment:
I'm a bit confused by this, is it safe to treat a global like this?

Reply:
In this case it's fine, as far as I understand `std::move` lets the compiler treat the variable with less concern about what happens to the source memory. For msvc after this function resolves the variable contains garbage. The next line then sets the `_itype` member to a value that means `player.HoldItem.isEmpty()` returns true, and most other normal uses of the HoldItem variable use this check to see if there is an item held. I say most, because some checks use `pcurs` instead which gets set by the call to `NewCursor()` two lines down.

Reply:
Thanks for the explanation, with the other cleanup this is also reduced to this case which is the more clean of the two that where there previously.